route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -43,35 +42,47 @@ namespace route_following_plugin
         plugin_discovery_msg_.capability = "strategic_plan/plan_maneuvers";
         pose_sub_ = nh_->subscribe("current_pose", 1, &RouteFollowingPlugin::pose_cb, this);
         twist_sub_ = nh_->subscribe("current_velocity", 1, &RouteFollowingPlugin::twist_cd, this);
-        
-        pnh_->param<double>("minimal_maneuver_duration", mvr_duration_, 16.0);
-        pnh2_->param<double>("config_speed_limit",config_limit);
-        pnh_->param<double>("buffer_time_lanechange",buffer_lanechange_time_);
-        pnh2_->param<double>("guidance/route_end_jerk", jerk_, 1.0);
-        pnh2_->param<std::string>("guidance/lane_change_plugin", lane_change_plugin_);
+
         wml_.reset(new carma_wm::WMListener());
         // set world model point form wm listener
         wm_ = wml_->getWorldModel();
-
+        //set a route callback to update route and calculate maneuver
+        wml_->setRouteCallback(route_set);
+        if(route_set)    
+        {
+            route_cb();
+        }
```

Comment:
I know what you are trying to do here, but this isn't the correct syntax. 
setRouteCallback() takes in a function which will be triggered when a new route has been received and processed by the world model. The way this is currently written route_set will be called every time a route is received which is probably an exception since its not a populated function object. What you are really looking for would be something like
 ```
 // Using std::bind to generate an std::function object from a function pointer
  wml_->setRouteCallback(std::bind(&RouteFollowingPlugin::route_cb, this));
 
 // OR
 
 // Using C++ lambda syntax
 wml_->setRouteCallback([this]() { route_cb(); });
 ```